### DISCUSSION TITLE
ContractMethodNoResultError

### DISCUSSION THREAD

**Discussion by nfts2me** [wevm/wagmi#2518] [General]
I'm getting this error.

However, instead of keeping the previous data, it returns null.

Is there a way to keep the previous data (if existed) instead of returning null when an error like this appear?

```
ContractMethodNoResultError: Contract read returned an empty response. This could be due to any of the following:
- The contract does not have the function "xxxx",
- The parameters passed to the contract function may be invalid, or
- The address is not a contract.
```

**Comment by tmm** [comment]
Can you share a minimal reproduction? Can't help unless you provide more info.

**Reply by nfts2me** [reply]
> Can you share a minimal reproduction? Can't help unless you provide more info.

Literally any contract which points to an unreliable RPC, like the one from Shardeum Sphnix (https://docs.shardeum.org/network/endpoints - , which is https://sphinx.shardeum.org/ )

Maybe a 20% of the requests return that instead of the actual value.

**Reply by tmm** [reply]
Are you using `wagmi` or `@wagmi/core`? What hooks/actions are you using? What chains are you using?

**Reply by nfts2me** [reply]
wagmi. Using "useContractReads" from wagmi 0.12.x

**Reply by tmm** [reply]
Can you create a minimal reproduction using [new.wagmi.sh](https://new.wagmi.sh)? That way I can open the link and immediate look into fixing instead of wasting time trying to recreate what you are seeing.
